### PR TITLE
Added ContainerClient.ExistsAsync() and BlobBaseClient.ExistsAsync()

### LINKED ISSUE #8952: [FEATURE] [storage] Implement a low performance cost Exists() method on the blobclient class
There is a very common use case to test if a blob exists before attempting to download it. This feature was supported in previous official Dotnet client libraries.  

e.g. CloudBlobClient.GetContainerReference(containerName).GetBlockBlobReference(key).Exists();

It is currently frustrating that the only solution is to drive logic flow with a try-catch which is bad practise and doesn't explicitly answer the question about whether the blob exists or not, just that it couldn't be fetched. We can look at status codes etc but the added complication increases branching logic and bug risk in the calling application. There is also the option of looking at the blob's properties but this has similar drawbacks to the above.   

Ideally, I would love an Exists() method on the BlobClient class, similar to previous versions of client libraries.  This should naturally be a low impact on performance and network traffic as is possible.

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/8952

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs [hunk 1]
```diff
@@ -931,6 +931,113 @@ private void SetNameFieldsIfNull()
...
+        /// </summary>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> if successful.
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: Should this say Response{bool} indicating whether the container exists or not?
Also similar comment for BlobBaseClient.

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs [hunk 2]
```diff
@@ -931,6 +931,113 @@ private void SetNameFieldsIfNull()
...
+        /// The <see cref="ExistsAsync"/> operation can be called on a
+        /// <see cref="BlobContainerClient"/> to see if the associated container
+        /// exists on the storage account in the storage service.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> if successful.
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: Should this say Response{bool} indicating whether the container exists or not?
Also similar comment for BlobBaseClient.

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: fixed.

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 3]
```diff
@@ -1935,6 +1935,114 @@ private void SetNameFieldsIfNull()
...
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<bool>> ExistsInternal(
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (Pipeline.BeginLoggingScope(nameof(BlobBaseClient)))
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs]: We should create a custom diagnostic scope specific to `Exists` - like you did for DataLake operations that wrapped other blob operations - so customers will be able to see the exists check in their distributed tracing.

**Pull request merged**